### ISSUE TITLE
Why is $toast undefined?

### ISSUE THREAD

**Issue by wsw70** [buefy/buefy#735]
### Overview of the problem

**Buefy** version: 0.6.5
**Vuejs** version: 2.5.13
**OS/Browser**: Windows 10 / Chrome 65

### Description

I would like to add a toast to my existing (and working) code

### Steps to reproduce

I have the current (working) method in my Vue instance

```
deleteEntry: (id) => {
            console.log(`deleting ${id}`)
        },
```

which I would like to enrich with a toast:

```
deleteEntry: (id) => {
            this.$toast.open({ message: `deleting ${id}`, type: "is-success" })
            console.log(`deleting ${id}`)
        },
```

### Expected behavior

The toast to pop-up

### Actual behavior

I get an `Uncaught TypeError: Cannot read property 'open' of undefined`. It seems that `$toast` is not defined - should there be anything special to do in order to have it defined?

- this is part of an otherwise working page (Buefy is defined and various components are used all over the page)
- the [example code](https://codepen.io/pen?editors=1010) works as well so there must be something missing in my code (?)

**Comment by adrlen** [comment]
Define your methods this way :

```
deleteEntry (id) {
     this.$toast.open({ message: `deleting ${id}`, type: "is-success" })
     console.log(`deleting ${id}`)
},
```
In your exemple, `this` refer to the internal scope of your method and not the component.

**Comment by adrlen** [comment]
More info : https://medium.freecodecamp.org/learn-es6-the-dope-way-part-ii-arrow-functions-and-the-this-keyword-381ac7a32881

**Comment by wsw70** [comment]
Thank you. I use JS casually from time to time and did read about fat arrows (that was the reason I started to use them). I was under the impression that the fact that I used them did exactly the opposite of what you said: **not** to have a new, local `this` (and address, say, vue `data` elements via `this.something` instead of going through some `that = this` monstrosities).
You are obviously right, I need to read more on the subject :)

Thanks again.
